### PR TITLE
- added Joi Validation

### PR DESCRIPTION
**Pull request by kabiakou**
- added Joi validator
- added additional GET request within Joi studying

### CODE REVIEW

File: js/validators/SchemaValidator.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const validate = (schema, property) => {
+    return (req, res, next) => {
+        console.log(req[property])
+        const { error } = schema.validate(req[property])
+        const valid = error == null
+        if (valid) {
```
Comment (m-bondarev) [inline js/validators/SchemaValidator.js]: I think you can simplify it like this:

```
        if (error) {
            const { details } = error;
            const message = details.map(i => i.message).join(',')
            res.status(400).json({
                code: 400,
                message: message
            })
        }
        
        next();
```

**Review by m-bondarev — APPROVED** [review]

File: js/validators/SchemaValidator.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            next()
+        } else {
+            const { details } = error;
+            const message = details.map(i => i.message).join(',')
+            res.status(400).json({
+                code: 400,
+                message: message
+            })
+        }
```
Comment (mdziarkach) [inline js/validators/SchemaValidator.js:11]: let's exclude `else` section
```js
if (valid) {
  next();
}

const { details } = error;
const message = details.map(i => i.message).join(',')
res.status(400).json({
  code: 400,
  message: message
})
```

File: js/validators/SchemaValidator.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        console.log(req[property])
+        const { error } = schema.validate(req[property])
+        const valid = error == null
+        if (valid) {
+            next()
+        } else {
+            const { details } = error;
+            const message = details.map(i => i.message).join(',')
+            res.status(400).json({
+                code: 400,
```
Comment (mdziarkach) [inline js/validators/SchemaValidator.js]: status already provided. Maybe you can remove `code` line?

File: js/validators/SchemaValidator.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const validate = (schema, property) => {
+    return (req, res, next) => {
+        console.log(req[property])
+        const { error } = schema.validate(req[property])
+        const valid = error == null
```
Comment (mdziarkach) [inline js/validators/SchemaValidator.js]:
```js
const valid = !!error
```

**Review by plutsenka — APPROVED** [review]

File: js/validators/SchemaValidator.js (hunk 4 above)
Comment (kabiakou) [reply js/validators/SchemaValidator.js]: Fixed

File: js/validators/SchemaValidator.js (hunk 3 above)
Comment (kabiakou) [reply js/validators/SchemaValidator.js]: Fixed

File: js/validators/SchemaValidator.js (hunk 2 above)
Comment (kabiakou) [reply js/validators/SchemaValidator.js:11]: Fixed

File: js/validators/SchemaValidator.js (hunk 1 above)
Comment (kabiakou) [reply js/validators/SchemaValidator.js]: Fixed

**Pull request merged**